Graphics/GraphicsEngine/interface/Shader.h

```diff
@@ -45,13 +45,16 @@ static const INTERFACE_ID IID_Shader =
 /// Describes the shader type
 DILIGENT_TYPED_ENUM(SHADER_TYPE, Uint32)
 {
-    SHADER_TYPE_UNKNOWN  = 0x000, ///< Unknown shader type
-    SHADER_TYPE_VERTEX   = 0x001, ///< Vertex shader
-    SHADER_TYPE_PIXEL    = 0x002, ///< Pixel (fragment) shader
-    SHADER_TYPE_GEOMETRY = 0x004, ///< Geometry shader
-    SHADER_TYPE_HULL     = 0x008, ///< Hull (tessellation control) shader
-    SHADER_TYPE_DOMAIN   = 0x010, ///< Domain (tessellation evaluation) shader
-    SHADER_TYPE_COMPUTE  = 0x020  ///< Compute shader
+    SHADER_TYPE_UNKNOWN       = 0x000, ///< Unknown shader type
+    SHADER_TYPE_VERTEX        = 0x001, ///< Vertex shader
+    SHADER_TYPE_PIXEL         = 0x002, ///< Pixel (fragment) shader
+    SHADER_TYPE_GEOMETRY      = 0x004, ///< Geometry shader
+    SHADER_TYPE_HULL          = 0x008, ///< Hull (tessellation control) shader
+    SHADER_TYPE_DOMAIN        = 0x010, ///< Domain (tessellation evaluation) shader
+    SHADER_TYPE_COMPUTE       = 0x020, ///< Compute shader
+    SHADER_TYPE_AMPLIFICATION = 0x040, ///< Amplification (task) shader
+    SHADER_TYPE_MESH          = 0x080, ///< Mesh shader
+    SHADER_TYPE_LAST
```

Comment:
SHADER_TYPE_LAST will be 0x081, which is not a flag.
Maybe use SHADER_TYPE_LAST = SHADER_TYPE_MESH, and then <= in the loop?

Reply:
The idea is make same as `Count`
```cpp
enum {
  Value1,
  Value2,
  Count
};
for (...; i < Count;) {}
```
and
```cpp
enum {
  Bit1 = 1 << 0,
  Bit2 = 1 << 1,
  Last // or BitCount ?
};
for (...; bit < Last;) {}
```